version: '2.1'

volumes:
  django_sample_db_dev: {}
  django_sample_packages: {}

services:
  db:
    image: postgres:9.6.1
    volumes:
      - django_sample_db_dev:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=primeword
      - POSTGRES_USER=futurer
      - POSTGRES_PASSWORD=futurer123
      - POSTGRES_INITDB_ARGS=--encoding=UTF-8
    ports:
      - "5432:5432"
    healthcheck:
      test: "pg_isready -h localhost -p 5432 -U postgres"
      interval: 3s
      timeout: 1s
      retries: 10

  django:
    build:
      context: .
#      dockerfile: ./compose/django/Dockerfile-dev
      dockerfile: ./Dockerfile
    environment:
      - DJANGO_DEBUG=True
      - DJANGO_DB_HOST=postgres
      - DJANGO_DB_PORT=5432
      - DJANGO_DB_NAME=primeword
      - DJANGO_DB_USERNAME=futurer
      - DJANGO_DB_PASSWORD=futurer123
      - DJANGO_SECRET_KEY=dev_secret_key
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy
    links:
      - db:postgres
    command: /start.sh
    volumes:
      - ./manage.py:/app/manage.py
      - ./requirements.txt:/app/requirements.txt
      - ./primeword_backend:/app/primeword_backend
      - ./word:/app/word
#      - ./restaurants:/app/restaurants
      - ./static:/app/static
      - ../gunicorn:/var/log/gunicorn
#      - django_sample_packages:/usr/local/lib/python3.6/site-packages/
#      - /usr/local/lib/python3.5/site-packages/rest_framework_swagger/static:/app/static
#      - /usr/local/lib/python3.5/site-packages/rest_framework/static:/app/static
#      - /usr/local/lib/python3.5/site-packages/django/contrib/admin/static:/app/static